spring-boot-project/spring-boot-docs/src/docs/antora/modules/reference/pages/testing/testcontainers.adoc

```diff
@@ -5,14 +5,69 @@ The https://www.testcontainers.org/[Testcontainers] library provides a way to ma
 It integrates with JUnit, allowing you to write a test class that can start up a container before any of the tests run.
 Testcontainers is especially useful for writing integration tests that talk to a real backend service such as MySQL, MongoDB, Cassandra and others.
 
+In following sections we will describe some of the methods you can use to integrate Testcontainers with your tests.
+
+== Using via @Testcontainers JUnit5 extension
+
+The Testcontainers framework provides JUnit5 extensions, which can be used to manage containers in your tests.
+The extension is activated by applying the
+
 Testcontainers can be used in a Spring Boot test as follows:
 
 include-code::vanilla/MyIntegrationTests[]
 
-This will start up a docker container running Neo4j (if Docker is running locally) before any of the tests are run.
+This will start up a Docker container running Neo4j (if Docker is running locally) before any of the tests are run.
 In most cases, you will need to configure the application to connect to the service running in the container.
 
+In this case the lifecycle of the container instance is managed by Testcontainers framework, as described in official documentation.
+
+== Using via Spring managed beans
+
+The containers provided by Testcontainers framework can be managed by Spring Boot as beans.
+This method is often used in combination with javadoc:org.springframework.boot.testcontainers.service.connection.ServiceConnection[format=annotation].
```

Comment:
`@ServiceConnection` can be used with static fields and beans.

Reply:
I do not quite understand what you mean with that, in context of the code which is selected here in comment. Is it that the sentence "This method is often used in combination with..." seems to point that it is the only combination in which `@ServiceConnection` annotation is used?

Reply:
So I rephrased the sentence so that it is clearer that this is not the only option. I hope now it is satisfactory. If not feel free to continue the discussion.